src/run_environment/common.ml

```diff
@@ -46,6 +46,86 @@ module Unique_id = struct
   include Ketrew_pure.Internal_pervasives.Unique_id
 end
 
+(** 
+
+Generate unique filenames for a given set of uniquely identifying properties.
+
+The module generates names that remain (way) under 255 bytes because they
+are the most common
+{{:https://en.wikipedia.org/wiki/Comparison_of_file_systems#Limits}file-system limits}
+in 2016.
+
+*)
+module Name_file = struct
+  (* Additional safety: we check that there are no hash-duplicates
+     within a given execution of a program linking with Biokepi. *)
+  let db : (string, string list) Hashtbl.t = Hashtbl.create 42
+
+  let path ~readable_suffix ~from high_level_components =
+    let sanitize =
+      String.map
+        ~f:(function
+          | ('a' .. 'z' | 'A' .. 'Z' | '0' .. '9' | '-') as c -> c
+          | other -> '_') in
+    let components =
+      begin match from with
+      | `Path p ->
+        let b = Filename.basename p in
+        (try [Filename.chop_extension b] with _ -> [b])
+      | `In_dir d -> []
+      end
+      @
+      List.map high_level_components ~f:sanitize
+    in
+    let hash =
+      String.concat ~sep:"-" (readable_suffix :: components)
+      |> Digest.string |> Digest.to_hex
+    in
+    let max_length = 220 in
```

Comment:
Ah right—might be worth noting this length in the documentation, then, since it's important for callers to know. (Could also be nice to make it an option, but not a big deal!)